centreon/www/include/configuration/configObject/servicegroup/formServiceGroup.php

```diff
@@ -34,124 +34,143 @@
 );
-$form->addElement('select2', 'sg_hServices', _("Linked Host Services"), array(), $attrService1);
+$form->addElement('select2', 'sg_hServices', _('Linked Host Services'), [], $attrService1);
 
-$route = './include/common/webServices/rest/internal.php?object=centreon_configuration_service' .
-    '&action=defaultValues&target=servicegroups&field=sg_hgServices&id=' . $sg_id;
+$route = './include/common/webServices/rest/internal.php?object=centreon_configuration_service'
+    . '&action=defaultValues&target=servicegroups&field=sg_hgServices&id=' . $serviceGroupId;
 $attrHostgroup1 = array_merge(
     $attrHostgroups,
-    array('defaultDatasetRoute' => $route)
+    ['defaultDatasetRoute' => $route]
 );
-$form->addElement('select2', 'sg_hgServices', _("Linked Host Group Services"), array(), $attrHostgroup1);
+$form->addElement('select2', 'sg_hgServices', _('Linked Host Group Services'), [], $attrHostgroup1);
 
-$route = './include/common/webServices/rest/internal.php?object=centreon_configuration_servicetemplate' .
-    '&action=defaultValues&target=servicegroups&field=sg_tServices&id=' . $sg_id;
+$route = './include/common/webServices/rest/internal.php?object=centreon_configuration_servicetemplate'
+    . '&action=defaultValues&target=servicegroups&field=sg_tServices&id=' . $serviceGroupId;
 $attrServicetemplate1 = array_merge(
     $attrServicetemplates,
-    array('defaultDatasetRoute' => $route)
+    ['defaultDatasetRoute' => $route]
 );
-$form->addElement('select2', 'sg_tServices', _("Linked Service Templates"), array(), $attrServicetemplate1);
+$form->addElement('select2', 'sg_tServices', _('Linked Service Templates'), [], $attrServicetemplate1);
 
-/*
- * Further informations
- */
-$form->addElement('header', 'furtherInfos', _("Additional Information"));
-$sgActivation[] = $form->createElement('radio', 'sg_activate', null, _("Enabled"), '1');
-$sgActivation[] = $form->createElement('radio', 'sg_activate', null, _("Disabled"), '0');
-$form->addGroup($sgActivation, 'sg_activate', _("Status"), '&nbsp;');
-$form->setDefaults(array('sg_activate' => '1'));
+// Further informations
+$form->addElement('header', 'furtherInfos', _('Additional Information'));
+$sgActivation[] = $form->createElement('radio', 'sg_activate', null, _('Enabled'), '1');
+$sgActivation[] = $form->createElement('radio', 'sg_activate', null, _('Disabled'), '0');
+$form->addGroup($sgActivation, 'sg_activate', _('Status'), '&nbsp;');
+$form->setDefaults(['sg_activate' => '1']);
+
+$form->addElement('textarea', 'sg_comment', _('Comments'), $attrsTextarea);
+
+if (
+    $o === SERVICE_GROUP_ADD
+    && $isCloudPlatform === true
+) {
+    $form->addElement(
+        'select2',
+        'resource_access_rules',
+        _('Resource Access Rule'),
+        [],
+        [
+            'datasourceOrigin' => 'ajax',
+            'availableDatasetRoute' => BASE_ROUTE . '?object=centreon_administration_aclgroup&action=list&for_cloud=true&all_servicegroups_filter=true',
```

Comment:
Specifying the environment type in the request parameter is not secure.
```suggestion
            'availableDatasetRoute' => BASE_ROUTE . '?object=centreon_administration_aclgroup&action=list&all_servicegroups_filter=true',
```

Reply:
Seen with callapa